pennylane/interfaces/tf.py

```diff
@@ -103,6 +110,9 @@ def grad(grad_output):
             else:
                 grad_input = tf.convert_to_tensor(grad_input)
 
+            if variables is not None:
+                return grad_input, variables
```

Comment:
A weird edge case that is related to tensorflow/tensorflow#31945, and caused by `@tf.custom_gradient` inspecting the function, and searching for all declared variables even if they are not passed as `grad_input`. Since these declared variables are not used across the Interface+QNode | device boundary, we can just return them with no manipulation.